### PR TITLE
eEvent overview staff page

### PR DESCRIPTION
**Pull request by feedmypixel**
This is a page that gives easy to check information on all Live events and Draft events associated with Membership. It can be found on ``/staff/event-overview``

- behind staff oauth,
- shows Live and Draft events and displays if they are a:
  - Free Event, 
  - No Discount Event, 
  - Not Sold through Eventbrite, 
  - Sold out

It also shows the:
  - Current main image,
  - Title of the event,
  - link to view in Ophan,
  - link to preview event,
  - link to edit page in eventbrite 

## Notes
- I have created a tools.scala.html file as we had a lot of markup in the main.scala.html that is not needed here. 
- This tools.scala.html brings in a separate stylesheet. There a few dependencies to be fixed in another pull request meaning I have had to bring in more than I need.

# Example - Events with Tags
screen shot 2015-01-21 at 11 57 50

# Desktop
event overview the guardian membership

# Mobile
event overview the guardian membership 1

@tackley for consistency the Ophan link now reads - "View in Ophan"

### CODE REVIEW

**Comment by davidrapson** [comment]
:+1:  for front-end code. It's a good move to extract the internal tools styling from the rest of the app. Keen to keep this stuff as much as possible. There's a lot of Scala in this PR so will let @wpf500 or @rtyley judge that.

**Comment by feedmypixel** [comment]
@jamesoram please have a look over

File: frontend/app/controllers/Staff.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import play.api.mvc.Controller
+import services.GuardianLiveEventService
+
+import scala.concurrent.Future
+
+trait Staff extends Controller {
+  val guLiveEvents = GuardianLiveEventService
+
+  def eventOverview = PermanentStaffNonMemberAction.async { implicit request =>
+     Future.successful(Ok(views.html.staff.eventOverview(guLiveEvents.events, guLiveEvents.eventsDraft)))
```
Comment (wpf500) [inline frontend/app/controllers/Staff.scala]: You don't need `Future.successful` here, just remove the `.async` on the `Action`

File: frontend/app/views/support/Event.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  case class EventAttributes(
+    free: Attribute,
+    noDiscount: Attribute,
+    notSoldThroughEventbrite: Attribute,
+    soldOut: Attribute
+  )
+
+  def eventDetail(event: EBEvent) = {
+      val free = !event.isNoTicketEvent && event.generalReleaseTicket.exists(_.free)
```
Comment (rtyley) [inline frontend/app/views/support/Event.scala]: I feel like a fair bit of this logic (`free` & `live`) could live against the EBEvent class itself - the other two fields, `notSoldThroughEventbrite` and `soldOut` already do.

File: frontend/app/views/support/Event.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      val noDiscount = !event.isNoTicketEvent && event.generalReleaseTicket.exists(!_.free && !event.hasMemberTicket)
+      val notSoldThroughEventbrite = event.isNoTicketEvent
+      val soldOut = event.isSoldOut
+
+    EventAttributes(
```
Comment (rtyley) [inline frontend/app/views/support/Event.scala]: Fairly sure that `EventAttributes` case class shouldn't exist - this method should just return a `Seq[Attribute]`.

File: frontend/app/views/support/Event.scala [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package views.support
+
+import model.Eventbrite.EBEvent
+
+object Event {
+
+  val ImportantClassName = "event-trait--important"
+  val ModerateClassName = "event-trait--moderate"
+  val NoteWorthyClassName = "event-trait--noteworthy"
```
Comment (wpf500) [inline frontend/app/views/support/Event.scala]: Not a massive fan on CSS classes in Scala, I think we could do something nicer here

**Comment by feedmypixel** [comment]
@jamesoram @rtyley @wpf500 changes as discussed added in.

@tackley the link name has changed it now says "ophan" as illustrated here: 
screen shot 2015-01-22 at 10 00 35
To give this context - this is behind staff oAuth, it is repeated a lot on the page (for every event) and we don't have much space. Please let me know if this is an issue.

**Comment by rtyley** [comment]
:+1:

**Comment by jamesoram** [comment]
:+1:

**Comment by rtyley** [comment]
I can see this on https://membership.theguardian.com/staff/event-overview - looks great!

Two minor flaws we can pick up later:

* It renders a little big on my screen, I can only see 8 events completely without scrolling! Would like to be able to see twice that many at least.
* Mixed content from using the insecure images, we should only ever use the secure ones!

**Comment by feedmypixel** [comment]
No probs :
- secure images being address [here](https://github.com/guardian/membership-frontend/pull/102)
- Will make smaller
- Also the ophan link needs updating as our urls have changed recently

**Comment by feedmypixel** [comment]
> Also the ophan link needs updating as our urls have changed recently

https://github.com/guardian/membership-frontend/pull/108

**Comment by feedmypixel** [comment]
> It renders a little big on my screen, I can only see 8 events completely without scrolling! Would like to be able to see twice that many at least.

this has been improved [in](https://github.com/guardian/membership-frontend/pull/111)

**Pull request merged**